kv/kvserver: TestStoreRangeMergeConcurrentRequests failed

kv/kvserver.TestStoreRangeMergeConcurrentRequests [failed](https://tanzanite.cluster.engflow.com/invocations/default/4fdd7866-a9cc-43a6-bc6e-4fdf98dffced?testReportRun=1&testReportShard=3&testReportAttempt=1#targets-Ly9wa2cva3Yva3ZzZXJ2ZXI6a3ZzZXJ2ZXJfdGVzdA==) on master @ [44de2d379610067e14a7ebfbc92e64311f13a232](https://github.com/cockroachdb/cockroach/commits/44de2d379610067e14a7ebfbc92e64311f13a232):


```
=== RUN   TestStoreRangeMergeConcurrentRequests
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestStoreRangeMergeConcurrentRequests980267701
    test_log_scope.go:76: use -show-logs to present logs inline
    client_merge_test.go:2378: context deadline exceeded
    panic.go:626: -- test log scope end --
test logs left over in: outputs.zip/logTestStoreRangeMergeConcurrentRequests980267701
--- FAIL: TestStoreRangeMergeConcurrentRequests (55.35s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=1</code>
 - <code>shard=3</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/kv
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestStoreRangeMergeConcurrentRequests.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

--- a/pkg/kv/kvserver/client_merge_test.go
+++ b/pkg/kv/kvserver/client_merge_test.go
@@ -2332,8 +2332,8 @@ func TestStoreRangeMergeConcurrentRequests(t *testing.T) {
 	}
 
 	// Failures in this test often present as a deadlock. Set a short timeout to
-	// limit the damage.
-	ctx, cancel := context.WithTimeout(ctx, testutils.DefaultSucceedsSoonDuration)
+	// limit the damage, but a longer deadline for race builds.
+	ctx, cancel := context.WithTimeout(ctx, testutils.SucceedsSoonDuration())
 	defer cancel()
 
 	const numGetWorkers = 16

